### PR TITLE
Support more callback in event listener

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/kvstore/EventListner.h [hunk 1]
```diff
@@ -13,31 +13,243 @@ namespace kvstore {
...
+                return "Compaction";
+            case rocksdb::BackgroundErrorReason::kWriteCallback:
+                return "WriteCallback";
+            case rocksdb::BackgroundErrorReason::kMemTable:
+                return "MemTable";
+        }
+        return "Unknown";
+    }
+
+    std::string writeStallConditionString(const rocksdb::WriteStallCondition& condition) {
```
Comment (critical27) [inline src/kvstore/EventListner.h:228]: Print the prev and cur condition in OnStallConditionsChanged

File: src/kvstore/EventListner.h [hunk 2]
```diff
@@ -13,31 +13,243 @@ namespace kvstore {
...
                   << ", compacted " << info.input_files.size()
                   << " files into " << info.output_files.size()
                   << ", base level is " << info.base_input_level
                   << ", output level is " << info.output_level;
     }
+
+    // A callback function to RocksDB which will be called
+    // before a RocksDB starts to flush memtables.
+    void OnFlushBegin(rocksdb::DB*, const rocksdb::FlushJobInfo& info) override {
+        LOG(INFO) << "Rocksdb start flush column family: " << info.cf_name
```
Comment (critical27) [inline src/kvstore/EventListner.h:42]: VLOG is enough for new added EventListener?

Comment (darionyaphet) [reply src/kvstore/EventListner.h:42]: Not enough. `Flush`  and `Ingest` should logged in INFO Level.

File: src/kvstore/EventListner.h (hunk 1 above)
Comment (darionyaphet) [reply src/kvstore/EventListner.h:228]: fixed

**Review by critical27 — APPROVED** [review]
LGTM

File: src/kvstore/EventListner.h [hunk 3]
```diff
@@ -13,31 +13,245 @@ namespace kvstore {
...
-        LOG(INFO) << "Rocksdb write stall condition changed, column family: " << info.cf_name
-                  << ", pre " << static_cast<int32_t>(info.condition.prev)
-                  << ", cur " << static_cast<int32_t>(info.condition.cur);
-    }
-
+    // A callback function to RocksDB which will be called before a RocksDB starts to compact.
     void OnCompactionBegin(rocksdb::DB*, const rocksdb::CompactionJobInfo& info) override {
-        LOG(INFO) << "Rocksdb compact column family: " << info.cf_name
-                  << " because of " << static_cast<int32_t>(info.compaction_reason)
+        LOG(INFO) << "Rocksdb start completion column family: " << info.cf_name
```
Comment (liuyu85cn) [inline src/kvstore/EventListner.h:18]: typo

**Review by liuyu85cn — APPROVED** [review]

**Review by critical27 — APPROVED** [review]
LGTM

**Pull request merged**